src/edu/csus/ecs/pc2/ui/SubmitSampleRunsPane.java

```diff
@@ -0,0 +1,877 @@
+        submitLanguages.setBounds(33, 191, 650, 32);
+        controlsPane.add(submitLanguages);
+        
+        checkBoxLanguage = new JCheckBox("Languages");
+        checkBoxLanguage.setVerticalAlignment(SwingConstants.TOP);
+        checkBoxLanguage.setToolTipText("Select Languages");
+        checkBoxLanguage.setSelected(true);
+        submitLanguages.add(checkBoxLanguage);
+        
+        submitAllLanguagesRadioButton = new JRadioButton("Submit All Languages");
+        submitAllLanguagesRadioButton.setSelected(true);
+        submitLanguages.add(submitAllLanguagesRadioButton);
+        
+        submitSelectedLanguagesRadioButton = new JRadioButton("Submit Selected");
+        submitLanguages.add(submitSelectedLanguagesRadioButton);
+        
+        selectedLanguagesLabel = new JLabel("None selected Lang?");
+        selectedLanguagesLabel.setToolTipText("None selected");
+        submitLanguages.add(selectedLanguagesLabel);
+        
+        selectLanguageButton = new JButton("Select");
+        selectLanguageButton.addActionListener(new ActionListener() {
+            public void actionPerformed(ActionEvent e) {
+                showLanguageList();
+            }
+        });
+        selectLanguageButton.setToolTipText("Select Problems");
+        submitLanguages.add(selectLanguageButton);
+        
+
+        languagesButtonGroup.add(submitAllLanguagesRadioButton);
+        languagesButtonGroup.add(submitSelectedLanguagesRadioButton);
+        
+        problemsButtonGroup.add(submitAllProblemsRadioButton);
+        problemsButtonGroup.add(submitSelectedProblemsRadioButton);
+        
+		VersionInfo info = new VersionInfo();
+		String verstring = info.getBuildNumber() + " x " + info.getPC2Version();
+		addLineToTextArea(verstring);
+		
+		
+		
+	}
+
+	protected void showLanguageList() {
+
+	    // TODO show language selection
+	    
+        Language[] Languages = getContest().getLanguages();
+        
+        if (Languages.length == 0) {
+            showMessage("No Languagess defined in contest");
+            return;
+        }
+    
+        JListFrame selectLanguagessFrame = new JListFrame("Select Languages ", Languages, selectedLanguagesIndexes, new ISelectedListsSetter() {
+            
+            @Override
+            public void setSelectedValuesList(List<Object> selectedValuesList, int[] selectedIndices) {
+                selectedLanguagesIndexes = selectedIndices;
+                updateLanguagesLabel(selectedValuesList);
+            }
+        });
+        selectLanguagessFrame.setVisible(true);
+    }
+ 
+
+    protected void showProblemList() {
+        
+        Problem[] problems = getContest().getProblems();
+        
+        if (problems.length == 0) {
+            showMessage("No problems defined in contest");
+            return;
+        }
+    
+        JListFrame selectProblemsFrame = new JListFrame("Select Problems ", problems, selectedProblemsIndexes, new ISelectedListsSetter() {
```

Comment:
See the comment below (in `showJudgementTypeList()`) about the window-close flaw in this method.

Reply:
I consider this an approval showstopper.